.github/workflows/test.yml

```diff
@@ -0,0 +1,34 @@
+name: Testing
+
+on: [push, pull_request]
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
+      - name: Install deb dependencies
+        run: sudo apt-get install -y xorg-dev libglu1-mesa-dev libglew-dev
+      - name: Install python dependencies
+        run: pip install -r requirements.txt
+      - name: Cache python wheel packages
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/pip
+          key: wheel-cache-${{ hashFiles('requirements.txt') }}
+      - name: Install evolution gym
+        run: pip install -e .
+      - name: Run test
+        run: python -c 'import evogym; print(f"{evogym.__name__} was loaded.")'
```

Comment:
Do you think this is sufficient as a test? Here is a more involved piece of code that we could potentially run. What are your thoughts on this? Maybe we could remove the `env.render()` line?

```
body, connections = sample_robot((5,5))
env = gym.make('Walker-v0', body=body)
env.reset()

for _ in range(100):
    action = env.action_space.sample()-1
    ob, reward, done, info = env.step(action)
    env.render() 

    if done:
        env.reset()

env.close()
```

Reply:
I don't think the current tests (import only) are sufficient. This can only confirm that the simulator was built.

The set of code snippet you suggested is more reinforced as a test, and I think it's better. I tried including this code and got a segmentation fault.

temporary branch and commit
https://github.com/TakesxiSximada/evogym/commit/b1c47213746f81e5092625aed4e9d348e4eb61aa

error output
https://github.com/TakesxiSximada/evogym/actions/runs/4642844324/jobs/8217059242#step:8:13

I think I might need Xvfb to do this on github actions and am investigating now. But I don't understand anything yet.

Reply:
Its possible that there is some sort of headless rendering issue? Maybe we can remove the `env.render() ` line? I wasn't able to understand much from the print message.

Reply:
> Its possible that there is some sort of headless rendering issue? 

There is no confirmation, but I think so.

I think it's reasonable to remove render from the test code. Because `env.render` function is gym module's function. (by unittest point of view)

From an integration testing point of view it also makes sense to do that(env.render).  env.render function is very slowly, but as long as it's only called once, I think it's acceptable. and I also found that it is possible with Xvfb (https://github.com/TakesxiSximada/evogym/commit/91386559296d223ce4a7547214ecc734483f769f).

No env.render (0.432s)
https://github.com/TakesxiSximada/evogym/actions/runs/4643005651/jobs/8217354081

Do env.render (3.986s)
https://github.com/TakesxiSximada/evogym/actions/runs/4643086514/jobs/8217492598

Based on the above, I think that installing Xvfb and running env.render as a test can contribute to better quality assurance.
I

Reply:
Awesome! I agree, if we can test render its preferred. Happy the Xvfb install fixed this!

Reply:
https://github.com/EvolutionGym/evogym/pull/25/commits/3da7f5c1d179447ee9086b42f98da6e6a360f5b7
https://github.com/EvolutionGym/evogym/pull/25/commits/080d8c139909bc612b10fc42d1ec2dab299bbd2e